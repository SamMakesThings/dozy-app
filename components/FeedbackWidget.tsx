import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  ViewProps,
  TouchableOpacity,
  TextInput,
} from 'react-native';
import { withTheme } from '@draftbit/ui';
import { scale } from 'react-native-size-matters';
import { Rating } from 'react-native-ratings';
import { Theme } from '../types/theme';

export interface FeedbackWidgetProps extends ViewProps {
  theme: Theme;
  rate: number;
  hasShadow?: boolean;
  onRateChange?: (rate: number) => void;
  onFeedbackChange?: (value: string) => void;
  onSubmit?: () => void;
}

export const FeedbackWidget: React.FC<FeedbackWidgetProps> = ({
  theme,
  style,
  rate,
  hasShadow = false,
  onRateChange,
  onFeedbackChange,
  onSubmit,
  ...props
}) => (
  <View
    style={[
      hasShadow && theme.elevation[3],
      styles.container,
      hasShadow && styles.containerShadow,
      {
        borderRadius: theme.borderRadius.global,
        backgroundColor: theme.colors.medium,
      },
      style,
    ]}
    {...props}
  >
    <Rating
      showRating
      fractions={1}
      ratingTextColor={theme.colors.primary}
      tintColor={theme.colors.medium}
      jumpValue={0.5}
      minValue={0.5}
      startingValue={0}
      onFinishRating={onRateChange}
    />
    <Text
      style={[
        theme.typography.headline6,
        { color: theme.colors.primary },
        styles.feedbackTitle,
      ]}
    >
      Please tell us why you gave this rating.
    </Text>
    <TextInput
      style={[
        styles.feedback,
        theme.typography.body2,
        {
          backgroundColor: theme.colors.background,
          color: theme.colors.strong,
        },
      ]}
      multiline
      numberOfLines={5}
      onChangeText={onFeedbackChange}
    />
    <TouchableOpacity
      style={[
        styles.button,
        { backgroundColor: theme.colors.primary },
        rate === 0 && styles.disabledButton,
      ]}
      disabled={rate === 0}
      onPress={onSubmit}
    >
      <Text
        style={[theme.typography.button, { color: theme.colors.secondary }]}
      >
        Submit
      </Text>
    </TouchableOpacity>
  </View>
);

const styles = StyleSheet.create({
  // eslint-disable-next-line react-native/no-color-literals
  container: {
    alignSelf: 'stretch',
    alignItems: 'center',
    padding: 20,
  },
  containerShadow: {
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.4,
    shadowRadius: 8,
    elevation: 3,
  },
  feedbackTitle: {
    marginTop: scale(16),
  },
  feedback: {
    alignSelf: 'stretch',
    height: scale(100),
    marginTop: scale(8),
    padding: scale(8),
    textAlignVertical: 'top',
    borderRadius: scale(8),
  },
  button: {
    marginTop: scale(24),
    paddingHorizontal: scale(16),
    paddingVertical: scale(8),
    borderRadius: scale(8),
  },
  disabledButton: {
    opacity: 0.5,
  },
});

export default withTheme(FeedbackWidget);
