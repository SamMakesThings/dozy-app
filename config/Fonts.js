export default {
  RubikRegular: require("../assets/fonts/RubikRegular.ttf"),
  RubikMedium: require("../assets/fonts/RubikMedium.ttf"),
  RubikBold: require("../assets/fonts/RubikBold.ttf")
}
