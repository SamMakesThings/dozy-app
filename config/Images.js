export default {
  Model016: require('../assets/images/model-016.png'),
  UndrawBed: require('../assets/images/UndrawBed.png'),
  WaveHello: require('../assets/images/WaveHello.png'),
  Slumberwordmark: require('../assets/images/Slumberwordmark.png'),
  DreamBg: require('../assets/images/DreamBg.png'),
  WordmarkTrans: require('../assets/images/WordmarkTrans.png'),
  DreamBgExtended: require('../assets/images/DreamBgExtended.png'),
  WomanInBed: require('../assets/images/WomanInBed.jpeg')
};
