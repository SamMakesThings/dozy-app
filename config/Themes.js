import { systemWeights } from 'react-native-typography';

export const slumber_theme = {
  disabledOpacity: 0.5,
  spacing: {
    gutters: 16,
    text: 4,
    small: 8,
    medium: 12,
    large: 16
  },
  borderRadius: {
    global: 6,
    button: 36
  },
  typography: {
    body1: {
      fontSize: 16,
      lineHeight: 26,
      fontFamily: 'RubikRegular'
    },
    body2: {
      fontSize: 14,
      lineHeight: 22,
      fontFamily: 'RubikRegular'
    },
    button: {
      fontSize: 16,
      lineHeight: 16,
      fontFamily: 'RubikRegular'
    },
    caption: {
      fontSize: 12,
      lineHeight: 16,
      fontFamily: 'RubikRegular'
    },
    headline1: {
      fontSize: 60,
      lineHeight: 71,
      fontFamily: 'RubikBold'
    },
    headline2: {
      fontSize: 48,
      lineHeight: 58,
      fontFamily: 'RubikMedium'
    },
    headline3: {
      fontSize: 38,
      lineHeight: 55,
      fontFamily: 'RubikMedium'
    },
    headline4: {
      fontSize: 24,
      lineHeight: 32,
      fontFamily: 'RubikBold'
    },
    headline5: {
      fontSize: 20,
      lineHeight: 26,
      fontFamily: 'RubikBold'
    },
    headline6: {
      fontSize: 16,
      lineHeight: 24,
      fontFamily: 'RubikBold'
    },
    overline: {
      fontSize: 12,
      lineHeight: 16,
      letterSpacing: 2,
      fontFamily: 'RubikRegular'
    },
    subtitle1: {
      fontSize: 16,
      lineHeight: 26,
      fontFamily: 'RubikRegular'
    },
    subtitle2: {
      fontSize: 14,
      lineHeight: 22,
      fontFamily: 'RubikRegular'
    }
  },
  colors: {
    background: 'rgba(35, 43, 63, 1)',
    divider: 'rgba(234, 237, 242, 1)',
    error: 'rgba(255, 69, 100, 1)',
    light: 'rgba(153, 172, 185, 1)',
    lightInverse: 'rgba(255, 255, 255, 0.68)',
    medium: 'rgba(64, 75, 105, 1)',
    mediumInverse: 'rgba(255, 255, 255, 0.87)',
    primary: 'rgba(0, 129, 138, 1)',
    secondary: 'rgba(219, 237, 243, 1)',
    strong: 'rgba(255, 255, 255, 1)',
    strongInverse: 'rgba(255, 255, 255, 1)',
    surface: 'rgba(255, 255, 255, 1)'
  },
  elevation: {
    '0': {
      shadowColor: 'rgba(255, 255, 255, 1)',
      shadowOffset: {
        width: 0,
        height: 0
      },
      shadowRadius: 0,
      shadowOpacity: 0,
      borderWidth: 0,
      borderColor: 'rgba(255, 255, 255, 1)',
      borderOpacity: 0
    },
    '1': {
      shadowColor: 'rgba(255, 255, 255, 1)',
      shadowOffset: {
        width: 0,
        height: 2
      },
      shadowRadius: 4,
      shadowOpacity: 0.06,
      borderWidth: 1,
      borderColor: 'rgba(255, 255, 255, 1)',
      borderOpacity: 0.06
    },
    '2': {
      shadowColor: 'rgba(255, 255, 255, 1)',
      shadowOffset: {
        width: 0,
        height: 2
      },
      shadowRadius: 4,
      shadowOpacity: 0.2,
      borderWidth: 0,
      borderColor: 'rgba(255, 255, 255, 1)',
      borderOpacity: 0
    },
    '3': {
      shadowColor: 'rgba(255, 255, 255, 1)',
      shadowOffset: {
        width: 0,
        height: 6
      },
      shadowRadius: 6,
      shadowOpacity: 0.12,
      borderWidth: 0,
      borderColor: 'rgba(255, 255, 255, 1)',
      borderOpacity: 0
    }
  }
};
