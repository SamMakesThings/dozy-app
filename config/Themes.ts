import { scale } from 'react-native-size-matters';
import { Theme } from '../types/theme';

// eslint-disable-next-line @typescript-eslint/naming-convention
export const dozy_theme: Theme = {
  disabledOpacity: 0.5,
  spacing: {
    gutters: 16,
    text: 4,
    small: 8,
    medium: 12,
    large: 16,
  },
  borderRadius: {
    global: 6,
    button: 36,
  },
  buttonLayout: {
    minHeight: scale(47),
    justifyContent: 'center',
  },
  typography: {
    smallLabel: {
      fontSize: scale(16),
      lineHeight: scale(26),
      fontFamily: 'RubikRegular',
    },
    body1: {
      fontSize: scale(18),
      lineHeight: scale(22),
      fontFamily: 'RubikRegular',
      letterSpacing: -0.25,
    },
    body2: {
      fontSize: scale(14),
      lineHeight: scale(22),
      fontFamily: 'RubikRegular',
    },
    button: {
      fontSize: scale(16),
      lineHeight: scale(16),
      fontFamily: 'RubikRegular',
    },
    caption: {
      fontSize: scale(12),
      lineHeight: scale(16),
      fontFamily: 'RubikRegular',
    },
    headline1: {
      fontSize: scale(60),
      lineHeight: scale(71),
      fontFamily: 'RubikBold',
    },
    headline2: {
      fontSize: scale(48),
      lineHeight: scale(58),
      fontFamily: 'RubikMedium',
    },
    headline3: {
      fontSize: scale(28),
      lineHeight: scale(35),
      fontFamily: 'RubikMedium',
    },
    headline4: {
      fontSize: scale(24),
      lineHeight: scale(32),
      fontFamily: 'RubikBold',
    },
    headline5: {
      fontSize: scale(22),
      lineHeight: scale(26),
      fontFamily: 'RubikMedium',
    },
    headline6: {
      fontSize: scale(16),
      lineHeight: scale(24),
      fontFamily: 'RubikMedium',
    },
    cardTitle: {
      fontSize: scale(20),
      lineHeight: scale(26),
      fontFamily: 'RubikMedium',
    },
    overline: {
      fontSize: scale(12),
      lineHeight: scale(16),
      letterSpacing: 2,
      fontFamily: 'RubikRegular',
    },
    subtitle1: {
      fontSize: scale(16),
      lineHeight: scale(26),
      fontFamily: 'RubikRegular',
    },
    subtitle2: {
      fontSize: scale(14),
      lineHeight: scale(22),
      fontFamily: 'RubikRegular',
    },
  },
  colors: {
    background: 'rgba(35, 43, 63, 1)',
    divider: 'rgba(234, 237, 242, 1)',
    error: 'rgba(255, 69, 100, 1)',
    light: 'rgba(153, 172, 185, 1)',
    lightInverse: 'rgba(153, 172, 185, 0.3)',
    medium: 'rgba(64, 75, 105, 1)',
    mediumInverse: 'rgba(255, 255, 255, 0.87)',
    primary: 'rgba(0, 129, 138, 1)',
    secondary: 'rgba(219, 237, 243, 1)',
    strong: 'rgba(255, 255, 255, 1)',
    strongInverse: 'rgba(255, 255, 255, 1)',
    surface: 'rgba(255, 255, 255, 1)',
  },
  elevation: {
    0: {
      shadowColor: 'rgba(18, 20, 44, 1)',
      shadowOffset: {
        width: 0,
        height: 0,
      },
      shadowRadius: 0,
      shadowOpacity: 0,
      borderWidth: 0,
      borderColor: 'rgba(18, 20, 44, 1)',
      borderOpacity: 0,
    },
    1: {
      shadowColor: 'rgba(18, 20, 44, 1)',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowRadius: 4,
      shadowOpacity: 0.06,
      borderWidth: 1,
      borderColor: 'rgba(18, 20, 44, 1)',
      borderOpacity: 0.06,
    },
    2: {
      shadowColor: 'rgba(18, 20, 44, 1)',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowRadius: 4,
      shadowOpacity: 0.2,
      borderWidth: 0,
      borderColor: 'rgba(18, 20, 44, 1)',
      borderOpacity: 0,
    },
    3: {
      shadowColor: 'rgba(18, 20, 44, 1)',
      shadowOffset: {
        width: 0,
        height: 6,
      },
      shadowRadius: 6,
      shadowOpacity: 0.12,
      borderWidth: 0,
      borderColor: 'rgba(18, 20, 44, 1)',
      borderOpacity: 0,
    },
  },
};
