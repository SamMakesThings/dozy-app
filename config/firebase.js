import * as firebase from 'firebase';
  
firebase.initializeApp(Expo.Constants.manifest.extra.firebase);

export default firebase;