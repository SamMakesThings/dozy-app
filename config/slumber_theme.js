import { systemWeights } from 'react-native-typography'

export const slumber_theme = {
  "disabledOpacity": 0.5,
  "spacing": {
    "gutters": 16,
    "text": 4,
    "small": 8,
    "medium": 12,
    "large": 16
  },
  "borderRadius": {
    "global": 3,
    "button": 24
  },
  "typography": {
    "body1": {
      "fontSize": 16,
      "lineHeight": 26,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "body2": {
      "fontSize": 14,
      "lineHeight": 22,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "button": {
      "fontSize": 14,
      "lineHeight": 16,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "caption": {
      "fontSize": 12,
      "lineHeight": 16,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "headline1": {
      "fontSize": 60,
      "lineHeight": 71,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "headline2": {
      "fontSize": 48,
      "lineHeight": 58,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "headline3": {
      "fontSize": 34,
      "lineHeight": 40,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "headline4": {
      "fontSize": 24,
      "lineHeight": 32,
      "letterSpacing": 0,
      ...systemWeights.semibold
    },
    "headline5": {
      "fontSize": 20,
      "lineHeight": 26,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "headline6": {
      "fontSize": 16,
      "lineHeight": 24,
      "letterSpacing": 0,
      ...systemWeights.semibold
    },
    "overline": {
      "fontSize": 12,
      "lineHeight": 16,
      "letterSpacing": 2,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "subtitle1": {
      "fontSize": 16,
      "lineHeight": 26,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "subtitle2": {
      "fontSize": 14,
      "lineHeight": 22,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    }
  },
  "colors": {
    "background": "rgba(35, 43, 63, 1)",
    "divider": "rgba(234, 237, 242, 1)",
    "error": "rgba(255, 69, 100, 1)",
    "light": "rgba(153, 172, 185, 1)",
    "lightInverse": "rgba(255, 255, 255, 0.68)",
    "medium": "rgba(64, 75, 105, 1)",
    "mediumInverse": "rgba(255, 255, 255, 0.87)",
    "primary": "rgba(0, 129, 138, 1)",
    "secondary": "rgba(219, 237, 243, 1)",
    "strong": "rgba(255, 255, 255, 1)",
    "strongInverse": "rgba(255, 255, 255, 1)",
    "surface": "rgba(255, 255, 255, 1)"
  },
  "elevation": {
    "0": {
      "shadowColor": "rgba(18, 20, 44, 1)",
      "shadowOffset": {
        "width": 0,
        "height": 0
      },
      "shadowRadius": 0,
      "shadowOpacity": 0,
      "borderWidth": 0,
      "borderColor": "rgba(18, 20, 44, 1)",
      "borderOpacity": 0
    },
    "1": {
      "shadowColor": "rgba(18, 20, 44, 1)",
      "shadowOffset": {
        "width": 0,
        "height": 2
      },
      "shadowRadius": 4,
      "shadowOpacity": 0.06,
      "borderWidth": 1,
      "borderColor": "rgba(18, 20, 44, 1)",
      "borderOpacity": 0.06
    },
    "2": {
      "shadowColor": "rgba(18, 20, 44, 1)",
      "shadowOffset": {
        "width": 0,
        "height": 2
      },
      "shadowRadius": 4,
      "shadowOpacity": 0.2,
      "borderWidth": 0,
      "borderColor": "rgba(18, 20, 44, 1)",
      "borderOpacity": 0
    },
    "3": {
      "shadowColor": "rgba(18, 20, 44, 1)",
      "shadowOffset": {
        "width": 0,
        "height": 6
      },
      "shadowRadius": 6,
      "shadowOpacity": 0.12,
      "borderWidth": 0,
      "borderColor": "rgba(18, 20, 44, 1)",
      "borderOpacity": 0
    }
  }
}