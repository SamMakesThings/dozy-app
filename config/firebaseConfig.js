import * as firebase from 'firebase';
import Expo from 'expo';
import Constants from 'expo-constants';
  
export const FbApp = !firebase.apps.length ? firebase.initializeApp(Constants.manifest.extra.firebase) : firebase.app();

export const FbAuth = firebase.auth();

export const FbLib = firebase;

// Making Firestore happen, testing data retrieval
/* 
var db = firebase.firestore();

var docRef = db.collection("cities").doc("SF");

docRef.get().then(function(doc) {
    if (doc.exists) {
        console.log("Document data:", doc.data());
    } else {
        // doc.data() will be undefined in this case
        console.log("No such document!");
    }
}).catch(function(error) {
    console.log("Error getting document:", error);
});
*/

// Google sign-in functionality
// const clientId = '713165282203-jjc54if1n7krahda9gvkio0siqltq57t.apps.googleusercontent.com';