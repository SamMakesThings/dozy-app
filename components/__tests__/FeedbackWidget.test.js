import React from 'react';
import { shallow } from 'enzyme';
import { TouchableOpacity } from 'react-native';
import { FeedbackWidget } from '../FeedbackWidget';
import { dozy_theme } from '../../config/Themes';

describe('FeedbackWidget', () => {
  it('should match the snapshot', () => {
    const component = shallow(<FeedbackWidget theme={dozy_theme} />);
    expect(component).toMatchSnapshot();
  });

  it('should call onSubmit function when rate is not 0', () => {
    const rate = 1;
    const onSubmit = jest.fn(() => rate);
    const component = shallow(
      <FeedbackWidget rate={rate} theme={dozy_theme} onSubmit={onSubmit} />,
    );

    expect(onSubmit).not.toHaveBeenCalled();

    component.find(TouchableOpacity).first().props().onPress();

    expect(onSubmit.mock.calls.length).toBe(1);
    expect(component.find(TouchableOpacity).first().props().onPress()).toBe(
      rate,
    );
  });

  it('should disable the submit button if rate is 0', () => {
    const rate = 0;
    const component = shallow(
      <FeedbackWidget rate={rate} theme={dozy_theme} />,
    );

    expect(
      component.find(TouchableOpacity).first().props().disabled,
    ).toBeTruthy();
  });
});
