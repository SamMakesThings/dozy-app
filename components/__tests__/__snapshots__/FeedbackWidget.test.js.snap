// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackWidget should match the snapshot 1`] = `
<View
  style={
    Array [
      false,
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "padding": 20,
      },
      false,
      Object {
        "backgroundColor": "rgba(64, 75, 105, 1)",
        "borderRadius": 6,
      },
      undefined,
    ]
  }
>
  <SwipeRating
    fractions={1}
    imageSize={40}
    jumpValue={0.5}
    minValue={0.5}
    ratingBackgroundColor="white"
    ratingColor="#f1c40f"
    ratingCount={5}
    ratingImage={1}
    ratingTextColor="rgba(0, 129, 138, 1)"
    showRating={true}
    showReadOnlyText={true}
    startingValue={0}
    tintColor="rgba(64, 75, 105, 1)"
    type="star"
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RubikMedium",
          "fontSize": 34.285714285714285,
          "lineHeight": 51.42857142857143,
        },
        Object {
          "color": "rgba(0, 129, 138, 1)",
        },
        Object {
          "marginTop": 34.285714285714285,
        },
      ]
    }
  >
    Please tell us why you gave this rating.
  </Text>
  <Component
    allowFontScaling={true}
    multiline={true}
    numberOfLines={5}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "borderRadius": 17.142857142857142,
          "height": 214.28571428571428,
          "marginTop": 17.142857142857142,
          "padding": 17.142857142857142,
          "textAlignVertical": "top",
        },
        Object {
          "fontFamily": "RubikRegular",
          "fontSize": 30,
          "lineHeight": 47.14285714285714,
        },
        Object {
          "backgroundColor": "rgba(35, 43, 63, 1)",
          "color": "rgba(255, 255, 255, 1)",
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <ForwardRef
    disabled={false}
    style={
      Array [
        Object {
          "borderRadius": 17.142857142857142,
          "marginTop": 51.42857142857143,
          "paddingHorizontal": 34.285714285714285,
          "paddingVertical": 17.142857142857142,
        },
        Object {
          "backgroundColor": "rgba(0, 129, 138, 1)",
        },
        false,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RubikRegular",
            "fontSize": 34.285714285714285,
            "lineHeight": 34.285714285714285,
          },
          Object {
            "color": "rgba(219, 237, 243, 1)",
          },
        ]
      }
    >
      Submit
    </Text>
  </ForwardRef>
</View>
`;
